lib/build.js

```diff
@@ -14,6 +14,8 @@ var fs = require('graceful-fs')
   , mkdirp = require('mkdirp')
   , exec = require('child_process').exec
   , win = process.platform == 'win32'
+  , semver = require('semver')
+  , runtime = semver.parse(process.version).major < 1 ? 'node' : 'iojs'
```

Comment:
in iojs you can check for `process.release.name`.
in nodejs `process.release` doesn't even exist, yet.

To set as a target I would recommend to use same notation as nvm: `gyp configure --target=iojs-v3.0.0`